File: compiled_lang/cpp/src/e40.branch.cpp
#include <iostream>

int main() {
    std::cout <<
        "Select an item from the menu.\n\n"
        "  1 - Coffee\n"
        "  2 - Espresso\n"
        "  3 - Latte\n"
        "  4 - Machiato\n"
        "  5 - Capucino\n"
        "  6 - Mocha\n"
        "  7 - Tea\n\n"
        "Make your selection: ";

    char keypress;
    std::cin.get(keypress);
    int selection = keypress - '0';

    switch (selection) {
        case 1: std::cout << "You selected a Coffee" << std::endl; break;
        case 2: std::cout << "You selected an Espresso" << std::endl; break;
        case 3: std::cout << "You selected a Latte" << std::endl; break;
        case 4: std::cout << "You selected a Machiato" << std::endl; break;
        case 5: std::cout << "You selected a Capucino" << std::endl; break;
        case 6: std::cout << "You selected a Mocha" << std::endl; break;
        case 7: std::cout << "You selected a Tea" << std::endl; break;
        default: std::cout << "You have not entered a valid selection" << std::endl;
    }

    return 0;
}

File: compiled_lang/cs/src/e40.branch.cs
using System;

class E40Branch
{
    static void Main()
    {
        Console.Write(
            "Select an item from the menu.\n\n" +
            "  1 - Coffee\n" +
            "  2 - Espresso\n" +
            "  3 - Latte\n" +
            "  4 - Machiato\n" +
            "  5 - Capucino\n" +
            "  6 - Mocha\n" +
            "  7 - Tea\n\n" +
            "Make your selection: ");

        int selection = Console.Read() - '0';

        switch (selection)
        {
            case 1:
                Console.WriteLine("You selected a Coffee");
                break;
            case 2:
                Console.WriteLine("You selected an Espresso");
                break;
            case 3:
                Console.WriteLine("You selected a Latte");
                break;
            case 4:
                Console.WriteLine("You selected a Machiato");
                break;
            case 5:
                Console.WriteLine("You selected a Capucino");
                break;
            case 6:
                Console.WriteLine("You selected a Mocha");
                break;
            case 7:
                Console.WriteLine("You selected a Tea");
                break;
            default:
                Console.WriteLine("You have not entered a valid selection");
                break;
        }
    }
}
